lib/eventsource.js

```diff
@@ -27,6 +27,19 @@ function hasBom (buf) {
   })
 }
 
+/**
+ * Wrap a callback to ensure it can only be called once.
+ */
+function once(cb) {
+  let called = false
+  return (...params) => {
+    if(!called) {
+      called = true
+      cb(...params)
+    }
```

Comment:
Is there a logger util in this library? It's helpful to see logs when something is ignored.

Reply:
There is not.